examples/vnet/doca_gw_dpdk.c

```diff
@@ -1,11 +1,42 @@
-#include "doca_gw.h"
-
 #include <stdio.h>
+#include "doca_gw_dpdk.h"
+
+
+struct gw_pipe_flow g_pipe_flow[MAX_PIP_FLOWS];
 
-int doca_gw_init(struct doca_gw_cfg *cfg,struct doca_gw_error *err)
+int doca_gw_init(struct doca_gw_cfg *cfg, struct doca_gw_error *err)
 {
-    printf("total sessions = %d\n",cfg->total_sessions);
-    return 0;
+	uint8_t idx;
+	struct gw_pipe_flow *pipe_flow;
+
+	err = err;
+	for(idx = 0 ; idx < MAX_PIP_FLOWS; idx++){
+		pipe_flow = &g_pipe_flow[idx];
+		memset(pipe_flow, 0x0, sizeof(struct gw_pipe_flow));
+		pipe_flow->item_param->item = &pipe_flow->items[idx];
+	}
+	printf("total sessions = %d\n",cfg->total_sessions);
+	return 0;
+}
+
+/**
+ * @brief get free pipeline flow from arry by used flage, currently.
+ *    
+ * @param match
+ *
+ * @return 
+ */
+static struct gw_pipe_flow* get_free_pipe_flow(void)
+{
+	uint8_t idx;
+	for(idx = 0 ; idx < MAX_PIP_FLOWS; idx++){
+		if (g_pipe_flow[idx].used == 0) {
+			g_pipe_flow[idx].used = 1;
+			return &g_pipe_flow[idx];
+		}
+	}
+	printf("no free pipe flow reserved.\n");
+	return NULL;
```

Comment:
running on arrays is bad practice. we can keep it for now but add //TODO change to O(1)